src/main/scala/viper/gobra/frontend/Source.scala

```diff
@@ -51,31 +51,39 @@ object Source {
   /**
    * Returns an object containing information about the package a source belongs to.
    */
-  def getPackageInfo(src: Source, projectRoot: Path): PackageInfo = {
-
-    val isBuiltIn: Boolean = src match {
-      case FromFileSource(_, _, builtin) => builtin
-      case _ => false
-    }
-
-    val packageName: String = PackageResolver.getPackageClause(src: Source)
-      .getOrElse(Violation.violation("Missing package clause in " + src.name))
-
-    /**
-     * A unique identifier for packages
-     */
-    val packageId: String = {
-      val prefix = uniquePath(TransformableSource(src).toPath.toAbsolutePath.getParent, projectRoot).toString
-      if(prefix.nonEmpty) {
-        // The - is enough to unambiguously separate the prefix from the package name, since it can't occur in the package name
-        // per Go's spec (https://go.dev/ref/spec#Package_clause)
-        prefix + " - " + packageName
-      } else {
-        // Fallback case if the prefix is empty, for example if the directory of a FileSource is in the current directory
-        packageName
+  def getPackageInfo(src: Source, projectRoot: Path): Either[Vector[ParserError], PackageInfo] = {
+    for {
+      packageName <- PackageResolver.getPackageClause(src: Source).toRight({
```

Comment:
do we need the type annotation `src: Source` here?

Reply:
I don't think so. No idea how that ended up here ^^